### ISSUE TITLE
Create an option for PG to create the folder structure required for ML or TPBF for any terrain parallax mod

### ISSUE THREAD

**Issue by hakasapl** [hakasapl/ParallaxGen#106]
From user `mutsisOnLihava` on Nexus:

Hi! I have a thought/suggestion for you.

So, if i have understood correctly parallaxGen upgrades "_p" to "_m", aka generates the complex material texture from parallax texture.

hmm.... There is a mod called "[Majestic landscapes](https://www.nexusmods.com/skyrimspecialedition/mods/41857)" that has a Patch called "[Majestic Landscapes - Enhanced Rocks and Mountains Patch And Other Fixes](https://www.nexusmods.com/skyrimspecialedition/mods/131170)". The idea of that patch is that it solves the "seams" problem between (mountain OR rock) AND terrain texture when using complex parallax textures. 
But there is a catch and its that the "[Majestic landscapes](https://www.nexusmods.com/skyrimspecialedition/mods/41857)﻿" have a(if i understood correct) specific folder structure for achieving blending, such that it uses copies of terrain blending textures(that are in special folder called "statics"),  that have white alpha(for blending purposes).

I recommend to read straight from the source("[Majestic landscapes](https://www.nexusmods.com/skyrimspecialedition/mods/41857)﻿" and its article and description AND "[Majestic Landscapes - Enhanced Rocks and Mountains Patch And Other Fixes](https://www.nexusmods.com/skyrimspecialedition/mods/131170)﻿" article), because i am not top tier explainer ':D.

Now to my idea: "Is it possible to generate this folder structure for all terrain texture packs, such that there exists white alpha channel in the texture, such that texture is in folder called statics".

This idea came up, cause even when "[Majestic Landscapes - Enhanced Rocks and Mountains Patch And Other Fixes](https://www.nexusmods.com/skyrimspecialedition/mods/131170)﻿" nearly solves the blending issue between terrain and rocks and "[Majestic landscapes](https://www.nexusmods.com/skyrimspecialedition/mods/41857)﻿" solves many other blending issues(its a base for the patch after all, not trying to steal credit from ML creator ':D), but the "problem"(well, not really a problem but in this case the thing trying to solve) is that because of this special folder structure and the need for white alpha channel in texture, it means that many texturepacks are not supporting the ML folder structure and white alpha channel solution.

So it would be handy, if FOR ANY texturepack there would be an algorithm(sort of like ParallaxGen) that would solve the white alpha channel problem for blending textures and the folder structure problem. Cause then, if one has ML and the patch,

1. one could use any landscape texture pack they want
2. Run the algorithm to generate folder structure and white alpha channel for blending texture
3. Have seamless landscape using any texturepack(mostly of course)

TLDR: Most texturepacks are incompatible with "[Majestic landscapes](https://www.nexusmods.com/skyrimspecialedition/mods/41857)﻿" folder structure and don't have the white alpha, so the seamless landscape cant be utilized because of compatibility issue.

**Comment by hakasapl** [comment]
Closing this as I made an SKSE plugin for this, just need ENB/CS to implement it: https://github.com/hakasapl/SkyrimTerrainSlotUnlocker
